Rubberduck.RegexAssistant/Atom.cs

```diff
@@ -0,0 +1,284 @@
+                    , HumanReadableClass());
+            }
+        }
+
+        private string HumanReadableClass()
+        {
+            return string.Join(", ", CharacterSpecifiers); // join last with and?
+        }
+
+        public bool TryMatch(ref string text)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is CharacterClass)
+            {
+                return (obj as CharacterClass).specifier.Equals(specifier);
+            }
+            return false;
+        }
+    }
+
+    class Group : Atom
+    {
+        public static readonly string Pattern = @"(?<!\\)\((?<expression>.*)(?<!\\)\)";
+        private static readonly Regex Matcher = new Regex("^" + Pattern + "$");
+
+        private readonly IRegularExpression subexpression;
+        private readonly string specifier;
+
+        public Group(string specifier) {
+            Match m = Matcher.Match(specifier);
+            if (!m.Success)
+            {
+                throw new ArgumentException("The given specifier does not denote a Group");
+            }
+            subexpression = RegularExpression.Parse(m.Groups["expression"].Value);
+            this.specifier = specifier;
+        }
+
+        public string Specifier
+        {
+            get
+            {
+                return specifier;
+            }
+        }
+
+        public string Description
+        {
+            get
+            {
+                return string.Format(AssistantResources.AtomDescription_Group, specifier) + "\r\n" + subexpression.Description;
+            }
+        }
+
+        public bool TryMatch(ref string text)
+        {
+            throw new NotImplementedException();
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (obj is Group)
+            {
+                return (obj as Group).specifier.Equals(specifier);
+            }
+            return false;
+        }
+    }
+
+    class Literal : Atom
+    {
+        public static readonly string Pattern = @"(?<expression>\\(u[\dA-F]{4}|x[\dA-F]{2}|[0-7]{3}|[bB\(\){}\\\[\]\.+*?1-9nftvrdDwWsS])|[^()\[\]{}\\*+?])";
+        private static readonly Regex Matcher = new Regex("^" + Pattern + "$");
+        private static readonly ISet<char> EscapeLiterals = new HashSet<char>();
+        private readonly string specifier;
+
+        static Literal() {
+            foreach (char escape in new char[]{ '.', '+', '*', '?', '(', ')', '{', '}', '[', ']', '|', '\\' })
+            {
+                EscapeLiterals.Add(escape);
+            }
+        }
+
+        public Literal(string specifier)
+        {
+            Match m = Matcher.Match(specifier);
+            if (!m.Success)
+            {
+                throw new ArgumentException("The given specifier does not denote a Literal");
+            }
+            this.specifier = specifier;
```

Comment:
The `this` qualifier is not needed (or it shouldn't be--if it is, something isn't right).